bedrock/firefox/views.py

```diff
@@ -351,17 +343,17 @@ def show_default_account_whatsnew(version):
 
 
 class FirstrunView(L10nTemplateView):
-    ftl_files_map = {
-        "firefox/firstrun/firstrun.html": ["firefox/firstrun"],
-        "firefox/developer/firstrun.html": ["firefox/developer"],
-    }
+    ftl_files_map = {"firefox/developer/firstrun.html": ["firefox/developer"], "firefox/new/desktop/download.html": ["firefox/new/desktop"]}
 
     def get(self, *args, **kwargs):
         version = self.kwargs.get("version") or ""
+        new_page_url = urlparams(reverse("firefox.new"), reason="outdated")
 
         # redirect legacy /firstrun URLs to /firefox/new/
-        if redirect_old_firstrun(version):
-            return HttpResponsePermanentRedirect(reverse("firefox.new"))
+        if detect_channel(version) != "developer":
+            return HttpResponsePermanentRedirect(new_page_url)
+        elif detect_channel(version) == "developer" and not show_57_dev_firstrun(version):
+            return HttpResponsePermanentRedirect(reverse("firefox.developer.index"))
```

Comment:
Suggestion: might it be worth creating a variable called `channel` so that you only need to call `detect_channel()` once? Something like

```python
channel = detect_channel(version)

if channel != "developer":
    return HttpResponsePermanentRedirect(new_page_url)
elif channel == "developer" and not show_57_dev_firstrun(version):
    return HttpResponsePermanentRedirect(reverse("firefox.developer.index"))